backend/substrapp/compute_tasks/outputs.py

```diff
@@ -54,64 +55,47 @@ def _save_performance(self, output: context.OutputResource):
             }
             client.register_performance(performance_obj)
 
-    def _save_models(self, models: list[context.OutputResource]):
-        logger.info("saving models")
+    @retry()
+    def _save_model_to_local_storage(self, models: list[context.OutputResource]) -> bytes:
+        url = settings.DEFAULT_DOMAIN + reverse("api:model-list")
+        files = {}
+        data = {}
 
-        new_models = []
         for model in models:
-            new_model = self._save_model_to_local_storage(model)
-            new_models.append(new_model)
-
-        try:
-            with get_orchestrator_client(self._ctx.channel_name) as client:
-                registered_models = client.register_models({"models": new_models})
-        except Exception as exc:
-            for model in new_models:
-                self._delete_model(model["key"])
-            raise exc
-
-        for registered_model in registered_models:
-            registered_model["channel"] = self._ctx.channel_name
-            serializer = ModelRepSerializer(data=registered_model)
-            try:
-                serializer.save_if_not_exists()
-            except AlreadyExistsError:
-                pass
-
-        for model, local in zip(models, new_models):
             path = os.path.join(self._ctx.directories.task_dir, model.rel_path)
-            add_model_from_path(path, local["key"])
+            checksum = get_hash(path, self._ctx.task.key)
```

Comment:
Why do you compute the checksum twice here and in the view ? I don't see how it could change.

Reply:
Yes, it's still a draft, I have to get it from Model output :)

Reply:
I re-check, I think it's just a git diff which is not clear, see the file https://github.com/Substra/substra-backend/blob/6f8b6be977c05acc561ec3226c45f94263cbcadd/backend/substrapp/compute_tasks/outputs.py
checksum is computed only one time. Let me know if you see twice computation !

Reply:
here: https://github.com/Substra/substra-backend/blob/6f8b6be977c05acc561ec3226c45f94263cbcadd/backend/substrapp/compute_tasks/outputs.py#L66
and here: https://github.com/Substra/substra-backend/blob/6f8b6be977c05acc561ec3226c45f94263cbcadd/backend/api/views/model.py#L102

Reply:
Yes, it's normal
The first one is before sending the model trough the API (from the worker) the other one is on the reception of the file by the server, we compare them in order to be sure that no data has been lost (corrupted file) https://github.com/Substra/substra-backend/blob/6f8b6be977c05acc561ec3226c45f94263cbcadd/backend/api/views/model.py#L106

Reply:
This is going to be really slow with all the I/Os and added network trips :/

Reply:
Yep, we can discuss about it as this PR is now functional, I will setup a meeting !

Reply:
I agree with @AlexandrePicosson here. HTTP ensures the integrity of the data transferred. Checking the checksum makes sense when retrieving data from cold storage, to ensure there was no corruption at rest. But here (in transit) I don't see much of a benefit since we have TCP already to make sure our data is correct (via the checksum feature of TCP).